### ISSUE TITLE
[Voice Access] Controls cannot be targeted by Voice Access when a tooltip is present

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#7815]
### Describe the bug

Voice access has a feature where each interactable control can have a number associated with it for targeting. In a XAML Islands app, if a tooltip is present, no controls will be tagged within the app. These number tags work as expected if a tooltip is not present/visible.

This bug does not repro on the WinUI 2 Gallery, making this a XAML Islands app.

Filing on behalf of the Windows Terminal team.

### Steps to reproduce the bug

1. Create a XAML islands app that has a button with a tooltip.
2. Open Voice Access
3. Say "Show numbers"
4. Observe that the button has a number tag
5. Say "Cancel"
6. Summon the tooltip by hovering over the button
7. Say "show numbers" again

### Expected behavior

The button should still have a number tag along with any other controls in the app.

### Screenshots

Unfortunately, summoning snipping tool removes the number tags. However, the accessibility team managed to get a screen capture of the issue in Windows Terminal.

[Tooltip.zip](https://github.com/microsoft/microsoft-ui-xaml/files/9760063/Tooltip.zip)

### Windows app type

- [ ] UWP
- [ ] Win32

### Device form factor

Desktop

### Windows version

Windows 11 (22H2): Build 22621

### Additional context

Using Microsoft.UI.Xaml 2.7.3-prerelease.220816001

**Comment by carlos-zamora** [comment]
This is still an issue

**Comment by damian-666** [comment]
** I really want to help on this feature as well as Copilot i've working on UI/API/NLS  in complex CAD for 35 years and started  in 90s .  and I'll try to find a An appropriate place to put holistic ideas later. If you want to just focus on this one bug, its a HUGE barrier  to use, it won't hurt to fix it. It's probably really easy. I. I'll try to about find  place to put the holistic advice later, test the new Copilot , etc.    THIS  why I'm unable to use this feature **hands free.** I have to still go over to the computer from across the room and press enter.

1.
launch Acrobat Reader . Manually for now, the start menu is another topic. just say" show numbers"  
and you're going to see numbers on all the buttons which don't have tool tips and those are the least important buttons. And there's no way to click them by voice . at least that I could get to work.


_I digress here. Which, by the way, "Show Numbers"  I would never have guessed existed. I'm not gonna ask remember, what can I say? or remember any of that -exactly- It's not that I'm stupid or disabled or lacking in a short term memory. It's that I'm thinking about quantum gravity right now. I don't have the bandwidth to think about a  user interface or any sets of commands about a learn cmds or mouse around giant menus.  I'm just gonna say what's intuitive, what I see,  Or what I or what I saw when I still could see, or  expect to see from on compiler. I'm going to expect opening files, building stuff. You know, I shouldn't even have to look at my screen. I might be cleaning my room. I mean, this is a huge feature. It's very important. This is the best thing you guys have ever done._

and you're going to see numbers on all the buttons which don't have tool tips and those are the least important buttons. And there's no way to click them. at least that I could get to work.  I also think it said at 123 you should put Alpha, Bravo, Charlie. Delta FoxTrot .  Port synonyms put a dictionary. when they say it highlighted and they can say cancel while it's still highlighted if they want. Bunt, like when you were using a mouse, you can click on it. You're like, oh, that's not the right thing. And then you move the mouse to the side and let go. That's bunting. There's a lot of issues here. You can replace mice and keyboard with this feature. I'm using a Shure cardoid  mic. And I'm having very good results with this. 

2.
I ran into this bug yesterday while experimenting with Bard, which is very interesting because it's using  the context in which you're speaking as a way to limit the domain that is uses to change voice to text to to make better guesses. LLM ,+ DSL Domain Specific Language. That's another issue that should probably go to the ai people.

the Bard mic button has tooltips so neither it nor windows can click it.   its not hands free and there's not really a "hotword" for Bard.   or a Send command.  button that because they have tooltips, i think.

#more off topic .. 
 But to just say how this bug affects it, I can't send. I can tell i  something and I can't say press send. If I say" show numbers" and say it, it's got a tool tip, so it's not going to work.   But what? was most frustrating is that Bard said that I could reassign hot words to the ui.  It's mic or send  button.  But even though it has agency or wants it. It really can't. It's lying. It says what you want to hear, or what it wants to do. It's very ambitious. I'm sure it's gonna succeed someday.   It can't. It doesn't have the power to press anything On the UI, but I do believe that it has some API access so that it could send itself. possibly a user query. I don't know. Well it it doesn't work because. Of the fact that that button has a tool tip, Now I know  so my next idea was to look at, say, press 
whatever tool tip said, but what it was telling me was that I could assign a custom hot word to the microphone button,
 
TLDR: #offtic , later , sorry.

  So I got very confused Yeah, you got to fix this. So im  talking to Bard, and I'm talking to Windows, and luckily it's not putting both in there you know, I think the Bard one has better potential for understanding exactly what I'm talking about because it's using context. But then again, I think yours is as well. But what it's not doing is using the context of the visual context around it. The api, the buttons, the names of them. It should definitely do that, especially when you're in a sort of a command mode. But that's another issue. You don't have to read all this. I'm going to put . 
 It has some agency, but it doesn't have that control. Windows should have that only. Bard, it actually shouldn't. I don't think it may be able to have some kind of plug in api that it can directly scrape via COM reflect, Silk Clang, and use that. But it. can't press a button. So it was telling me I said, well, I want to change the hot word from. sand to something, you know. To enter or send it or make it so. And it said, oh sure. Every time you press that then say those words and I'll learn to associate it, and then you'll change it.
 
 2.  Launch a browser and bring up Bard.   I press it, it starts listening. There's no hot word associated with that button, but I'm using Windows voice assist to invoke it. So I have both running, but Bard told me that I can say the  name of the Mic button ,   to invoke it make it start listening. 
 
 And or say press the microphone, by saying I want to call it "Mother " and it said I should say mother and then click it and it'll make that association and it'll learn a new name for itself. So this is a little bit to do with the feature that would be hot words and that is a little bit on aside. I'm just going to leave it here. This is all I have to say about this very specific bug. 
 
 I think you should fix this first and we can get into the other stuff later. II I've had, I have a few things to say that's tangential. I don't I know it might be against the guidelines. I'll paste it somewhere else if I just don't have time. I'm trying to get these scientists to use  Windows and to learn how to make simulations using chatgp. One is Prize Nobel Prize winning quantum gravity theorist who predicted LLMs , and they don't know how to do numerical simulations, some can't type or code, those that do likely use Linux , Julia and JustSayIt.iL . . So it's really important what you guys are doing and you can nail this. You can have Star Trek level UI, you're pretty close to it.  Look, I don't have time, but now I can move this to discussion or somebody else can copy paste it. I just really tired. I've been working 20 hours a day.
 
 
  but I don't know if it's lying or not. I do expect it can run its own API. I don't know. But what's interesting about Bard is that it's using the context of what we're talking about to help guess what I might be saying to get voice to guest better for Speech to Text, in situ . You're gonna be talking about the application that's in focus. object model I just had the start menu pop up, by the way, by I don't know what I said. . And it's because right now I'm talking to a specific app. If I want to talk to windows, I should just say, hey, Windows or start, annoying but still minor realignment.
  
  This is touching a lot of issues, but really, I'm digressing again. Just fix this bug for us and we can get a little bit further. 
  
  Use  Not only the context of what is being has or been said, but the context of the API's and where you're at. Have you just launched the app then you're probably going to say File View, etc. 
  
  I want to call an app on my Personal Computer anything I want." Hal ","Computer" or whatever to get its attention. But I think that there should be a hot word should be treated differently than general text because they they require constant monitoring and they can be less consuming of power if they're restricted in domain, 
   which is another issue. 
   
Maybe this is a little bit against the um contribution guidelines. I don't know if this is open source, but I'm I'm willing to help for free. I just can't comply with every little open. source 's. Convention. I'm not. I don't know. I'm a professional. II don't do open source. I'm I'm working for free right now for for quantum physicists. I, I, I don't have time to log a bug. I'm notAQA person. I'm an engineer. I make software. I might have repeated myself. I hope I haven't wasted your time, but I can save you months of work and make sure that this is a delightful feature and that you won't have to have a lot of meetings. It could set the model for every app and it could be the new start menu that that used to be in Windows 10. That work that didn't bring up Bing. every app could have a ribbon with a microphone on it. On the top left it's going to be wicked. It could be the best ui ever. I mean, everyone's going to want to use it. Doesn't matter how much of a mess. That design by consensus UI is it's always like that. It's it is, it's it's duplicate it's redundant, but after a while I mean you're going to discover it and then you're just going to speak to it and you never and you're going to command and query it. And then and then you'll never have to see it again.

**Comment by kmahone** [comment]
This is potentially a bug in Voice Access. I have filed a bug internally on the Voice Access team with this information.

**Comment by codendone** [comment]
A fix for this has been checked in for a future Windows release/update. ([internal tracking bug](http://task.ms/48914366))
